addons/survey/static/src/question_page/question_page_one2many_field.js

```diff
@@ -83,10 +86,22 @@ class QuestionPageOneToManyField extends X2ManyField {
             if (params.record) {
                 params.record = record.data[name].records.find(r => r.resId === params.record.resId);
             }
+            if (params.context.question_create) {
+                const limit = this.list.config.limit;
+                const offset = (Math.ceil(this.list.count / limit) - 1) * limit;
+                await this.list.load({ limit: limit, offset: offset });
+            }
             await openRecord(params);
         };
         this.canOpenRecord = true;
     }
+
+    async onAdd({ context, editable } = {}) {
+        const limit = this.list.config.limit;
+        const offset = (Math.ceil(this.list.count / limit) - 1) * limit;
+        await this.list.load({ limit: limit, offset: offset });
+        await super.onAdd({ context, editable });
+    }
```

Comment:
As mentioned above, this override doesn't seem required anymore with the fixup you have added as the last page is already loaded by the change in the fixup (if (params.context.question_create) { await this.loadLastPage(); }).